solutions/usaco-838.mdx

```diff
@@ -2,23 +2,105 @@
 id: usaco-838
 source: USACO Gold 2018 Open
 title: Milking Order
-author: Qi Wang, Melody Yu
+author: Qi Wang, Melody Yu, Neo Wang
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_milkorder_gold_open18.html)
 
 ## Video Solution
 
 Note: The video solution might not be the same as other solutions. Code in C++.
+
 <Youtube id="OaL9vEkShyk" />
 
+## Implementation
+
 **Time Complexity:** $\mathcal{O}((E + N \log N) \log M)$
 
+We can binary search on the last valid value of $M$ such that the ordering can
+be satisfied. For each iteration of the binary search, we only check if the
+interval of times $[0, x)$ is valid. Finally, to
+find the lexicographically smallest TopoSort, we use a `priority_queue` to process
+the ids from least to greatest.
+
 <LanguageSection>
 
-<JavaSection>
+<CPPSection>
+
+```cpp
+//CodeSnip{CPP Short Template}
+//BeginCodeSnip{Topological Sort}
+// /**
+//  * Description: sorts vertices such that if there exists an edge x->y, then x goes before y
+//  * Source: KACTL
+//  * Verification: https://open.kattis.com/problems/quantumsuperposition
+//  */
```

Comment:
1. what's this for
2. why do you have a `//` and a `/*`

Reply:
1. it's a description of the toposort whcih is warranted
2. this is fine. what's the issue?

Reply:
> this is fine. what's the issue?

it's just not necessary, as `/*...*/` already comments the code out